### DISCUSSION TITLE
Installed .net 6.0.400 on raspian - how to run MinecraftClient ?

### DISCUSSION THREAD

**Discussion by r3spawned** [ORelio/Minecraft-Console-Client#2136] [Q&A] [answered]
Hi,

I'm using a raspberry pi 400, os : Raspbian bullseye arm64 (64 bit version)

The latest version of Mcc doesn't seem to support mono any longer..  I see this message appear when running mono MinecraftClient.exe - "Cannot open assembly 'MinecraftClient.exe': File does not contain a valid CIL image.".

As per recommendation (found in the forum) I installed .net 6.0.400 but I have no idea how to launch MinecraftClient using it.

Mono was easy to use in linux.  I simply typed Mono MinecraftClient.exe to launch the program.  Using .net is new to me.

I would appreciate if someone could share basic step by step instructions what I have to do with the MinecraftClient file in order to get it working with .net 6 in linux.  Thankyou.

![mono no longer works](https://user-images.githubusercontent.com/51679834/187247960-d26f84c2-7715-4dad-85ed-8aed014c29fa.png)
![dotnet version](https://user-images.githubusercontent.com/51679834/187247966-5c320467-2af5-43d3-94a1-b16935e04eb7.png)

**Comment by breadbyte** [comment, marked as answer]
Get the linux build of the latest release, extract it, and simply run the linux executable on the terminal.

Here is a video demonstration:

https://user-images.githubusercontent.com/14045257/187249283-4383355f-4995-4982-be46-8ff8483a3cd5.mp4

**Reply by milutinke** [reply]
> Get the linux build of the latest release, extract it, and simply run the linux executable on the terminal.
> 
> Here is a video demonstration:
> v1.mp4

But that is an x86 version, he is running on ARM 64.

@r3spawned You will have to compile it yourself, I made a detailed manual in the documentation [here](https://mccteam.github.io/docs/guide/installation.html#linux-macos)
Just use `-r linux-arm64` in the build command instead of `-r linux-x64`

**Reply by breadbyte** [reply]
> But that is an x86 version, he is running on ARM 64.

I failed to notice, good catch. We should add a precompiled binary for ARM64.

**Reply by milutinke** [reply]
> > But that is an x86 version, he is running on ARM 64.
> 
> I failed to notice, good catch. We should add a precompiled binary for ARM64.

Ping me once you did it, so I can update the wiki.

**Reply by r3spawned** [reply]
> Get the linux build of the latest release, extract it, and simply run the linux executable on the terminal.
> 
> Here is a video demonstration:
> 
>  v1.mp4

I followed your video, changed the wget link to https://github.com/MCCTeam/Minecraft-Console-Client/releases/download/20220830-35/MinecraftClient-linux-arm64.zip from here : https://github.com/MCCTeam/Minecraft-Console-Client/releases/tag/20220830-35

It worked first time.  Easy to follow instructions.  Thanks so much!

_(Thanks to milutinke and breadbyte and anyone else who was involved in mentioning / adding arm64 version)_  

:)
